camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeperPlanner.java

```diff
@@ -0,0 +1,133 @@
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormatter;
+
+import com.linkedin.camus.sweeper.utils.DateUtils;
+
+
+public class CamusHourlySweeperPlanner extends CamusSweeperPlanner {
+  private static final String CAMUS_HOURLY_SWEEPER_MAX_HOURS_AGO = "camus.hourly.sweeper.max.hours.ago";
+  private static final String DEFAULT_CAMUS_HOURLY_SWEEPER_MAX_HOURS_AGO = "1";
+  private static final String CAMUS_HOURLY_SWEEPER_MIN_HOURS_AGO = "camus.hourly.sweeper.min.hours.ago";
+  private static final String DEFAULT_CAMUS_HOURLY_SWEEPER_MIN_HOURS_AGO = "1";
+
+  private static final Logger LOG = Logger.getLogger(CamusHourlySweeperPlanner.class);
+
+  private DateTimeFormatter hourFormatter;
+  private DateUtils dUtils;
+
+  @Override
+  public CamusSweeperPlanner setPropertiesLogger(Properties props, Logger log) {
+    dUtils = new DateUtils(props);
+    hourFormatter = dUtils.getDateTimeFormatter("YYYY/MM/dd/HH");
+    return super.setPropertiesLogger(props, log);
+  }
+
+  private DateTime getFolderHour(Path datePath, Path inputDir) {
+    String datePathStr = datePath.toString();
+    String inputDirStr = inputDir.toString();
+    String dateStr = datePathStr.substring(datePathStr.indexOf(inputDirStr) + inputDirStr.length());
+    return hourFormatter.parseDateTime(dateStr.replaceAll("^/", ""));
+  }
+
+  @Override
+  public List<Properties> createSweeperJobProps(String topic, Path inputDir, Path outputDir, FileSystem fs)
+      throws IOException {
+    return createSweeperJobProps(topic, inputDir, outputDir, fs, new CamusSweeperMetrics());
+  }
+
+  @Override
+  protected List<Properties> createSweeperJobProps(String topic, Path inputDir, Path outputDir, FileSystem fs,
+      CamusSweeperMetrics metrics) throws IOException {
+    LOG.info("creating hourly sweeper job props: topic=" + topic + ", inputDir=" + inputDir + ", outputDir="
+        + outputDir);
+    List<Properties> jobPropsList = new ArrayList<Properties>();
+    if (!fs.exists(inputDir)) {
+      LOG.warn("inputdir " + inputDir + " does not exist. Skipping topic " + topic);
+      return jobPropsList;
+    }
+
+    for (FileStatus f : fs.globStatus(new Path(inputDir, "*/*/*/*"))) {
+
+      DateTime folderHour = getFolderHour(f.getPath(), inputDir);
+      if (shouldProcessHour(folderHour)) {
+        Properties jobProps = createJobProps(topic, f.getPath(), folderHour, outputDir, fs, metrics);
+        if (jobProps != null) {
+          jobPropsList.add(jobProps);
+        }
+      }
+    }
+
+    return jobPropsList;
+  }
+
+  private Properties createJobProps(String topic, Path folder, DateTime folderHour, Path outputDir, FileSystem fs,
+      CamusSweeperMetrics metrics) throws IOException {
+
+    Properties jobProps = new Properties(props);
```

Comment:
Should create an empty Properties using `new Properties()`. The single parameter is used for default.